infra/build/functions/ci.py

```diff
@@ -0,0 +1,260 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+"""Tool for testing changes to base-images in OSS-Fuzz. This script builds test
+versions of all base images and the builds projects using those test images."""
+import argparse
+import collections
+import logging
+import sys
+
+from googleapiclient.discovery import build as cloud_build
+import oauth2client.client
+import requests
+
+import build_and_push_test_images
+import build_and_run_coverage
+import build_project
+
+IMAGE_PROJECT = 'oss-fuzz'
+BASE_IMAGES_PROJECT = 'oss-fuzz-base'
+TEST_IMAGE_SUFFIX = 'testing'
+FINISHED_BUILD_STATUSES = ('SUCCESS', 'FAILURE', 'TIMEOUT', 'CANCELLED',
+                           'EXPIRED')
+
+
+class ProjectStatus:
+  """Class that holds info about project builds."""
+
+  def __init__(self, name):
+    self.name = name
+    self.production_build = {'coverage': None, 'fuzzing': None}
+    self.build_result = {'coverage': None, 'fuzzing': None}
+    self.build_finished = {'coverage': True, 'fuzzing': True}
+    self.build_id = {'coverage': None, 'fuzzing': None}
+
+  def set_build_id(self, build_id, build_type):
+    """Sets the build id of |build_type| to |build_id|."""
+    self.build_id[build_type] = build_id
+    if build_id:
+      self.build_finished[build_type] = False
+
+  def set_build_result(self, result, build_type):
+    """Sets the result of |build_type| to |result|."""
+    self.build_result[build_type] = result
+    self.build_finished[build_type] = True
+
+
+def _get_production_build_statuses(build_type, statuses=None):
+  """Gets the statuses for |build_type| that is reported by build-status.
+  |statuses| is an optional dictionary mapping projects to dictionaries
+  containing the status of coverage and fuzzing builds. If provided, the
+  dictionary is appended to, otherwise a new dictionary is created and
+  returned."""
+  if statuses is None:
+    statuses = {}
+  if build_type == 'fuzzing':
+    filename = 'status.json'
+  elif build_type == 'coverage':
+    filename = 'status-coverage.json'
+  else:
+    assert None
```

Comment:
Rather than do this, raise a ValueError with an explanation.

Reply:
Since this is a programmer error, I think an assertion makes sense.
I've gotten rid of the else clause and added `assert build_type in ('fuzzing', 'coverage')` before the if-elif

Reply:
ValueErrors are idiomatic for programmer errors.

e.g. Python itself will raise ValueError for  things like:

```
>>> int('123g')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '123g'
```

Reply:
Done.